src/platform/NodePlatformFunctions.ts

```diff
@@ -119,6 +126,28 @@ class NodePlatformFunctions extends PlatformFunctions {
         });
     });
   }
+
+  /** @override */
+  createNodeHttpClient(agent: http.Agent): typeof HttpClient {
+    // @ts-ignore
+    return new NodeHttpClient(agent);
+  }
+
+  /** @override */
+  createHttpClient(agent?: http.Agent): typeof HttpClient {
```

Comment:
We shouldn't have the `http.Agent` parameter here. It's a node specific type.

Reply:
This is in `NodePlatformFunctions` - is the concern because the web implementation also inherits from `PlatformFunctions` which takes `agent?: http.Agent | null`?

Reply:
For `createDefaultHttpClient` the concern is that `agent?: http.Agent | null` doesn't apply to all implementations. it's ok to keep it on the createNodeHttpClient method where it makes sense.

Reply:
Fixed in latest commit as discussed!